spec/lib/jobs/generate-enclosure-spec.js

```diff
@@ -318,6 +329,115 @@ describe(require('path').basename(__filename), function () {
+            var node = {
+                id: nodeId,
+                relations: []
+            };
+            var enclId = uuid.v4();
+            var encl = [
+                {
+                    id: enclId,
+                    name: 'Enclosure Node ABC123',
+                    type: 'enclosure',
+                    relations: []
+                },
+            ];
+
+            var enclRelation = {
+                relations: [{
+                    relationType: 'encloses',
+                    targets: [nodeId]
+                }]
+            };
+
+            this.sandbox.stub(mockWaterline.nodes,'create');
+            this.sandbox.stub(mockWaterline.nodes,'find').resolves(encl);
+            this.sandbox.stub(mockWaterline.nodes,'findByIdentifier').resolves(node);
+            return job.run()
+            .then(function() {
+                expect(mockWaterline.nodes.create).to.not.have.been.called;
+                expect(mockWaterline.nodes.updateByIdentifier)
+                    .to.have.been.calledWith(enclId, enclRelation);
+            });
+        });
+
+        it('Should not override other relations when update node', function() {
+            var node = {
+                id: nodeId,
+                relations: [
+                    {
+                        relationType: 'poweredBy',
+                        targets: 'bbb'
+                    }
+                ]
+            };
+            var enclId = uuid.v4();
+            var encl = [
+                {
+                    id: enclId,
+                    name: 'Enclosure Node ABC123',
+                    type: 'enclosure',
+                    relations: [
+                        {
+                           relationType: 'poweredBy',
+                           targets: 'aaa'
+                        }
+                    ]
+                },
+            ];
+            var enclRelation = _.cloneDeep(encl[0].relations);
+            enclRelation.push({
+                relationType: 'encloses',
+                targets: [nodeId]
+            });
+            var nodeRelation = _.cloneDeep(node.relations);
+            nodeRelation.push({
+                relationType: 'enclosedBy',
+                targets: [enclId]
+            });
+
+            this.sandbox.stub(mockWaterline.nodes,'create');
+            this.sandbox.stub(mockWaterline.nodes,'find').resolves(encl);
+            this.sandbox.stub(mockWaterline.nodes,'findByIdentifier').resolves(node);
+            return job.run()
+            .then(function() {
+                expect(mockWaterline.nodes.create).to.not.have.been.called;
+                expect(mockWaterline.nodes.updateByIdentifier)
+                    .to.have.been.calledWith(enclId, {relations: enclRelation});
+                expect(mockWaterline.nodes.updateByIdentifier)
+                    .to.have.been.calledWith(nodeId, {relations: nodeRelation});
```

Comment:
Since the final node content is determined by external promise "updateByIdentifier", there is no convenient way to verify its output. The node's content is changed by this function, whose input has been verified by these two statements above to make sure only the (encloses-enclosedBy) relation will be modified, and it is reliable to pass this test case. Any thoughts?